Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP2StateMachine.swift

```diff
@@ -41,17 +41,22 @@ extension HTTPConnectionPool {
         private var state: State = .running
 
         init(
-            idGenerator: Connection.ID.Generator
+            idGenerator: Connection.ID.Generator,
+            maximumConcurrentHTTP1Connections: Int
         ) {
             self.idGenerator = idGenerator
             self.requests = RequestQueue()
 
             self.connections = HTTP2Connections(generator: idGenerator)
+            self.http1Connections = HTTP1Connections(
+                maximumConcurrentConnections: maximumConcurrentHTTP1Connections,
+                generator: idGenerator
+            )
```

Comment:
I bet we reserve an array within this initializer. Maybe we want to add another initializer here, that really creates an empty HTTP1Connections. The struct is replaced in the migration anyway.

The constructor should set `maximumConcurrentConnections` to a default value. Then we don't need to pass it in the HTTP2 constructor.